be/src/Project/Services/Catalog/Catalog.Api/Services/Categories/CategoriesService.cs

```diff
@@ -294,38 +300,45 @@ public async Task<CategorySchemaServiceModel> UpdateCategorySchemaAsync(UpdateCa
 
             this.TriggerCategoryProductsIndexRebuild(new RebuildCategoryProductsIndexServiceModel
             {
-                CategoryId = model.CategoryId,
+                CategoryId = model.Id,
                 Language = model.Language,
                 OrganisationId = model.OrganisationId,
                 Username = model.Username
             });
 
-            return await GetCategorySchemaAsync(new GetCategorySchemaServiceModel 
-            { 
-                CategoryId = model.CategoryId,
+            return GetCategorySchemas(new GetCategorySchemasServiceModel
+            {
+                Id = model.Id,
                 Language = model.Language,
                 OrganisationId = model.OrganisationId,
                 Username = model.Username
-            });
+            }).Result;
```

Comment:
**suggestion:** Get rid of .Result instead of await because GetCategorySchemas method is asynchronous